packages/dashboard/src/components/composed/RPCs/RPC/Details/Expanded/DecodedParams.tsx

```diff
@@ -0,0 +1,49 @@
+import { Group, Text, createStyles } from "@mantine/core";
+import { Prism } from "@mantine/prism";
+import { AlertTriangle } from "react-feather";
+import type { DecodableRpcMethod } from "src/utils/constants";
+
+const useStyles = createStyles((theme, _params, _getRef) => ({
+  alert: {
+    color: theme.colors.orange[6]
+  }
+}));
+
+type DecodedParamsProps = {
+  decodingInspected: string;
```

Comment:
Should we pass actual Decoding objects instead of `string`? Like, seems like this should probably be a [`Codec.CalldataDecoding`](https://trufflesuite.com/docs/truffle/codec/modules/_truffle_codec.html#calldatadecoding), and then we just convert to string as-needed.

Seems like this would prevent things from breaking in the future. Strings are dumb.

Reply:
Hm I figured it's nice to `import inspect from "object-inspect"` once, `inspect` once and pass the string around.

Like we won't do anything with the decoding other than `inspect`ing it before codec-components, and codec-components will replace much of this stuff anyway (it consumes `Codec.CalldataDecoding` and that's it, no need to pass that off to other components). Is this lazy?

Reply:
Don't know about lazy, but it definitely seems to make sense to pass around the raw thing rather than the thing that's missing information

Reply:
Updated to pass around `decoding` (typed `CalldataDecoding | string`, `string` because it might be a signature request, not `eth_sendTransaction`)

Reply:
Ah yeah hrm, that's a bit tricky. The `string` here strikes me as a code smell... ideally I wish it were easy to make EIP-712 signatures the responsibility of the codec (cc @haltman-at), but this doesn't really make sense.

Maybe we should wrap `string` in some Dashboard-defined type? That way the type gets a proper name, at least.

Reply:
Makes sense. I replaced `Calldata | string` with a `Decoding` type defined [here](https://github.com/trufflesuite/truffle/blob/e8618ce9a68eca4240d95d327b531b08c73d86e1/packages/dashboard/src/utils/dash.ts#L134).